src/models/continue_watching_preview.rs

```diff
@@ -1,44 +1,107 @@
-use crate::constants::CATALOG_PREVIEW_SIZE;
-use crate::models::common::eq_update;
-use crate::models::ctx::Ctx;
-use crate::runtime::msg::{Internal, Msg};
-use crate::runtime::{Effects, Env, UpdateWithCtx};
-use crate::types::library::{LibraryBucket, LibraryItem};
 use lazysort::SortedBy;
 use serde::Serialize;
 
+use crate::{
+    constants::CATALOG_PREVIEW_SIZE,
+    models::{common::eq_update, ctx::Ctx},
+    runtime::{
+        msg::{Internal, Msg},
+        Effects, Env, UpdateWithCtx,
+    },
+    types::{
+        library::{LibraryBucket, LibraryItem},
+        notifications::NotificationsBucket,
+    },
+};
+
+/// The continue watching section in the app
 #[derive(Default, Clone, Serialize, Debug)]
 #[serde(rename_all = "camelCase")]
-/// The continue watching section in the app
 pub struct ContinueWatchingPreview {
     pub library_items: Vec<LibraryItem>,
 }
 
 impl ContinueWatchingPreview {
-    pub fn new(library: &LibraryBucket) -> (Self, Effects) {
+    pub fn new(library: &LibraryBucket, notifications: &NotificationsBucket) -> (Self, Effects) {
         let mut library_items = vec![];
-        let effects = library_items_update(&mut library_items, library);
+        let effects = library_items_update(&mut library_items, library, notifications);
         (Self { library_items }, effects.unchanged())
     }
 }
 
 impl<E: Env + 'static> UpdateWithCtx<E> for ContinueWatchingPreview {
     fn update(&mut self, msg: &Msg, ctx: &Ctx) -> Effects {
+        // update the CW list if
         match msg {
-            Msg::Internal(Internal::LibraryChanged(_)) => {
-                library_items_update(&mut self.library_items, &ctx.library)
+            // library has changed
+            Msg::Internal(Internal::LibraryChanged(_))
+            // LibraryItem has been updated (this message alters the `mtime` and will re-order the CW list)
+            | Msg::Internal(Internal::UpdateLibraryItem(_))
```

Comment:
This might not be ideal, because it does update mtime, however, it does so even if you mark a movie or a video (episode) as watched.

- [ ] Discuss with @jaruba whether this is intended consequence